app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/TransportationDetailsActivity.kt

```diff
@@ -0,0 +1,167 @@
+import de.tum.`in`.tumcampusapp.component.other.general.model.Recent
+import de.tum.`in`.tumcampusapp.component.other.generic.activity.ProgressActivity
+import de.tum.`in`.tumcampusapp.component.ui.transportation.model.efa.Departure
+import de.tum.`in`.tumcampusapp.component.ui.transportation.model.efa.StationResult
+import de.tum.`in`.tumcampusapp.database.TcaDb
+import de.tum.`in`.tumcampusapp.utils.NetUtils
+import de.tum.`in`.tumcampusapp.utils.Utils
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+/**
+ * Activity to show transport departures for a specified station
+ *
+ *
+ * NEEDS: EXTRA_STATION set in incoming bundle (station name)
+ */
+class TransportationDetailsActivity : ProgressActivity(R.layout.activity_transportation_detail) {
+
+    private lateinit var mViewResults: LinearLayout
+    private lateinit var recentsDao: RecentsDao
+    private lateinit var transportManager: TransportController
+    private lateinit var gson: Gson
+
+    private val disposable = CompositeDisposable()
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // get all stations from db
+        recentsDao = TcaDb.getInstance(this).recentsDao()
+        transportManager = TransportController(this)
+        gson = Gson()
+        mViewResults = this.findViewById(R.id.activity_transport_result)
+
+        val intent = intent
+        if (intent == null) {
+            finish()
+            return
+        }
+        val location = intent.getStringExtra(EXTRA_STATION)
+        title = location
+        val locationID = intent.getStringExtra(EXTRA_STATION_ID)
+
+        showLoadingStart()
+        loadDetails(location, locationID)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.menu_transport, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (item.itemId == R.id.action_transport_usage) {
+            AlertDialog.Builder(this)
+                    .setTitle(R.string.transport_action_usage)
+                    .setMessage(R.string.transport_help_text)
+                    .setPositiveButton(android.R.string.ok, null)
+                    .create()
+                    .show()
+            return true
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun loadDetails(location: String, locationID: String) {
+        // Quality is always 100% hit
+        val stationResult = StationResult(location, locationID, Integer.MAX_VALUE)
+        val jsonStationResult = gson.toJson(stationResult)
+
+        // save clicked station into db
+        recentsDao.insert(Recent(jsonStationResult, RecentsDao.STATIONS))
+
+        // Check for internet connectivity
+        if (!NetUtils.isConnected(this)) {
```

Comment:
This check is performed in the `ConnectivityInterceptor`. You can remove it here.